mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -201,6 +213,78 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
       }
     }
 
+    int64_t gemmM_size, gemmN_size, gemmK_size;
+    int64_t gemmM_extra, gemmN_extra, gemmK_extra;
+    gemmM_extra = gemmN_extra = gemmK_extra = 0;
+    // compute we should use extra padding kernel or not
+    if (convOpType == miopen::ConvOpType::Conv2DOpType) {
+      gemmM_size = k / g;
+      gemmK_size = (c / g) * y * x;
+      gemmN_size = n * ho * wo;
```

Comment:
@asleepzzz are these equations layout-agnostic?

@asroy could you help check these equations? from the look of it, they seem correct to me but I want a second pair of eye go over them.